cocotb/share/include/gpi.h

```diff
@@ -72,7 +72,32 @@ we have to create a process with the signal on the sensitivity list to imitate a
 # define __attribute__(x)
 #endif
 
-
+/*
+ * Declare the handle types.
+ *
+ * We want these handles to be opaque pointers, since their layout is not
+ * exposed to C. We do this by using incomplete types. The assumption being
+ * made here is that `sizeof(some_cpp_class*) == sizeof(some_c_struct*)`, which
```

Comment:
You could actually assert for that.

Reply:
I couldn't, because what I need to assert is that the compiler behaves the same way in C and C++ mode. I could compromize and hard-code a size in, and assert it at both ends.

Reply:
... or only add the assertion in C++ mode. But don't worry, I don't expect to stay this code around forever anyways.